src/bank.rs

```diff
@@ -637,6 +638,14 @@ impl Bank {
         false
     }
 
+    pub fn hash_bank_state(&self) -> Hash {
```

Comment:
How about the name `hash_internal_state()`? And add some documentation about its intended use.